src/Microsoft.Data.Analysis/StringDataFrameColumn.BinaryOperations.cs

```diff
@@ -101,7 +101,7 @@ public PrimitiveDataFrameColumn<bool> ElementwiseEquals(string value)
             return ret;
         }
 
-        public override PrimitiveDataFrameColumn<bool> ElementwiseEquals<T>(T value)
+        public override PrimitiveDataFrameColumn<bool> ElementwiseValueEquals<T>(T value)
```

Comment:
How come it is `AddValue` but here it is `ValueEquals`?

Reply:
It feels like it should be `EqualsValue`, because it is a method on this column, and you are asking if this column equals a value. Not "value equals".

---
In reply to: [356132434](https://github.com/dotnet/corefxlab/pull/2801#discussion_r356132434) [](ancestors = 356132434)

Reply:
Yea I debated between the names. I know they are inconsistent but they seemed to make more sense to me. How does `ElementwiseNotEqualsValue` sound? I thought `ElementwiseValueNotEquals` sounded better. Similarly, `AddValue` sounded better than `ValueAdd`.

Reply:
Take "elementwise" out of the method name and just think about "Equals".

`myColumn.Equals(5)` is what we want to write.

Why does `myColumn.ValueEquals(5)` sound better than `myColumn.EqualsValue(5)`?  Especially if you think `AddValue` sounds better than `ValueAdd`.

Reply:
I think for the `Equals` method, `Value` can go in the front or the back and it still reads nicely. For `NotEquals` though, `ElementwiseValueNotEquals` almost sounds like a proper sentence. `ElementwiseNotEqualsValue` sounds like `Value` has been added to the end as an afterthought? Even if you disregard `Elementwise`, `ValueNotEquals` reads "better" than `NotEqualsValue` right?

Reply:
I'm not sure I agree. Because this is a method on a column. The code reads like English - "column not equals value 5" - `column.NotEqualsValue(5)`. I wouldn't say "column value not equals 5". Just like I wouldn't say "column value add 5". I would say "column add value 5".

Reply:
If instead we would pick the other overload (the one that takes a column), how would we do the naming?

Would we name it `column.ElementwiseColumnNotEquals(otherColumn)`?  Or would we name it `column.ElementwiseNotEqualsColumn(otherColumn)`.

In the end, I really don't want `Value` in the method name at all. But if we need to add it, I want it to be consistent on where/how we add `Value` so it makes sense as a whole. `AddValue`, `SubtractValue`, `NotEqualsValue`, etc.

Reply:
I agree that `Value` shouldn't be in the method name either. I modified the PR to remove all the `Value` additions to the APIs. For all the overloaded methods, we now check if the value passed in is a `DataFrameColumn` and call the right overload if it is.